raven/raven.go

```diff
@@ -268,3 +306,29 @@ func (T *transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	defer timer.Stop()
 	return T.httpTransport.RoundTrip(req)
 }
+
+type EventEncoder interface {
```

Comment:
Valid concerns. I think the best reason for making this pluggable is that it makes it easier to test components separately, the stacktrace-tests shouldn't be dependent on the encoding to work etc. If we used httptest we would need to have a lot of more "stuff" in the test that's not really related. We would also need to base64/zip-decode things before testing the value. Both approaches have their merits.

Regarding exportable I'm totally with you, it's just me not being 100% comfortable with the casing-decides-visibility yet :)